### ISSUE TITLE
[syntax sugar]Add support of hashtable

### ISSUE THREAD

**Issue by luodanwg** [neo-project/neo-devpack-dotnet#12]
### Abstract

This is  syntax sugar.with support of hashtable seems much convenience to save/read in smartcontract.

### Motivation

I think use hashTable to save/read  is much convenience , this can reduce the complexity of contract development and easier to understand

### Specification
this is the sample,  if with no hashTable, we have to 
``` c#
        public static object Main(string operation, params object[] args)
        {
            switch (operation)
            {
                ...
            }
        }

        private static byte[] Query(string domain)
        {
            return Storage.Get(Storage.CurrentContext, domain);   <=== hard to read and coding
        }

        private static bool Register(string domain, byte[] owner)
        {
            Storage.Put(Storage.CurrentContext, domain, owner);  <=== hard to read and coding
            return true;
        }

```
if we have hashTable ,we can use hashTable name ='domainTable' as the prefix to save/read the contract data.
``` c#
        public static object Main(string operation, params object[] args)
        {
            Hashtable domainTable=new Hashtable(); 
            switch (operation)
            {
                ...
            }
        }

        private static byte[] Query(string domain)
        {
            return domainTable[domain]    <=== easy to use and understand
        }

        private static bool Register(string domain, byte[] owner)
        {
            domainTable[domain]=owner   <=== easy to use and understand
            return true;
        }

```

### Rationale

na

### Backwards Compatibility

na

### Test Cases

na

### Implementation

na

**Comment by erikzhang** [comment]
I think the easiest way to use this syntax is:
```c#
StorageContext context = Storage.CurrentContext;
var xxx = context["Hello"];
```
Or just:
```c#
var xxx = Storage.CurrentContext["Hello"];
```

**Comment by lightszero** [comment]
i had add a support
https://github.com/neo-project/neo-devpack-dotnet/pull/13

**Comment by luodanwg** [comment]
thanks @erikzhang  & @lightszero for the solution of variable storage .
one more thing, we now should get the result in []byte first and convert it to expected types by manual(such as int => AsBigInteger() ) . can we define some Built-in data types. so the data conversion can become automatic. no need to use `AsBigInteger()` any more.

example:
``` c#
 //get(read),  key(string) -> value
 var xxx string = Storage.CurrentContext["Stored_key1_string_type"] 
 var yyy bigint 
 yyy = Storage.CurrentContext["Stored_key2_string_type"]

 //put(store),  key(string->"NewValue") ->value (string)
 var toBeStore string
 toBeStore = "should be store"
 Storage.CurrentContext["NewValue"] = toBeStore 
``` 

another requirement ,
I thinks we still have the requirement of define a map, like example below.
the map name `userBalance` will be the storage prefix, so all the value will store  under this prefix. and could allow define the key type and value type.
``` c#
var userBalance[addr]bigint
var admin address
var ceo address= "AVhvC8E257QXHmg66BQfXmfTZTbHyBpM6v"
var cfo address ="AHZKMK2AKxyAc7HQ381U649F8DybAh2Z3d"

userBalance[admin] =10000000
userBalance[ceo] = 1000
userBalance[cfo] = 500

func transfer(addr address, amount bigint){
          userBalance[addr] = userBalance[addr]  - amount
}
```

**Comment by lightszero** [comment]
one more thing use another issue please.
and we can't implement on neovm now.

**Comment by erikzhang** [comment]
I think it's possible to implement a storage based hashtable.
```c#
userBalance["admin"] = 10000;
```
This can be convert into:
```c#
Storage.Put(context, "userBalance" + "admin", 10000);
```

**Comment by lightszero** [comment]
last transaction will leave some keys if use storage for map,then we need more code to clean keys.

**Comment by erikzhang** [comment]
It is implemented in d1980fc448d178b88ab57fb5abf151215f67cb59
